### PR TITLE
test(ToggleButton): use react-testing-library instead of enzyme

### LINKED ISSUE #272: Migrate <ToggleButton /> test component from enzyme to react-testing-library

### PR DESCRIPTION
**Pull request by spartDev**
- [ ] Feature
- [ ] Fix
- [x] Enhancement

## Description
Refactor`<ToggleButton /> unit test with [react-testing-library](https://github.com/kentcdodds/react-testing-library) instead of [Enzyme](https://github.com/airbnb/enzyme/). 

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.

close #272

### CODE REVIEW

File: src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap [hunk 1]
```diff
@@ -13,8 +13,8 @@ exports[`<ToggleButton /> should call componentDidUpdate and setState 1`] = `
   width: 3.8rem;
   height: 2rem;
   border-radius: 1rem;
-  background: hsl(200,74%,46%) linear-gradient( 0deg, hsla(0,100%,100%,0) 0%, hsla(0,100%,100%,0.1) 100% );
-  border: 1px solid hsl(200,74%,46%);
+  background: hsl(207,22%,90%);
```
Comment (AntoninMarchardDev) [inline src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap]: Why does the linear gradient missing ?

File: src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap [hunk 2]
```diff
@@ -34,169 +34,17 @@ exports[`<ToggleButton /> should call componentDidUpdate and setState 1`] = `
   -webkit-transition: -webkit-transform 250ms ease;
   -webkit-transition: transform 250ms ease;
   transition: transform 250ms ease;
-  -webkit-transform: translateX(1.7rem);
-  -ms-transform: translateX(1.7rem);
-  transform: translateX(1.7rem);
+  -webkit-transform: translateX(0);
```
Comment (AntoninMarchardDev) [inline src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap]: Is that normal that some components are different ?

File: src/ToggleButton/index.js [hunk 3]
```diff
@@ -77,7 +77,7 @@ class ToggleButton extends PureComponent {
     return (
       <div className={className} onClick={() => this.handleToogle(checked)}>
         <Checkbox type="checkbox" checked={checked} readOnly />
-        <Toggle checked={checked} readOnly={disabled} />
+        <Toggle checked={checked} readOnly={disabled} data-testid="toggle" />
```
Comment (AntoninMarchardDev) [inline src/ToggleButton/index.js]: change `toggle` for `toggle-button`, we'll probably need `toggle` for something else later

File: src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap (hunk 2 above)
Comment (spartDev) [reply src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap]: @AntoninMarchardDev Can you be more explicit I don't understand?

Comment (AntoninMarchardDev) [reply src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap]: @spartDev `translateX` change from `translateX(1.7rem)` in the original snapshot to `translateX(0)` in the new snapshot is that normal ?

File: src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap (hunk 1 above)
Comment (AntoninMarchardDev) [reply src/ToggleButton/__tests__/__snapshots__/ToggleButton.test.js.snap]: @spartDev same here, the background hsl value is not the same between the old snapshot and the new

**Pull request merged**